apps/token-manager/app/src/components/LocalIdentityBadge/LocalIdentityBadge.js

```diff
@@ -0,0 +1,86 @@
+import React from 'react'
+import PropTypes from 'prop-types'
+import styled from 'styled-components'
+import { Badge, IdentityBadge, font } from '@aragon/ui'
+import { LocalIdentityModalContext } from '../LocalIdentityModal/LocalIdentityModalManager'
+import { IdentityContext } from '../IdentityManager/IdentityManager'
+
+const LocalIdentityBadge = ({ address, ...props }) => {
+  const { resolve } = React.useContext(IdentityContext)
+  const { showLocalIdentityModal, updates$ } = React.useContext(
+    LocalIdentityModalContext
+  )
+  const [label, setLabel] = React.useState()
+  const handleResolve = async () => {
+    try {
+      const { name = null } = await resolve(address)
+      setLabel(name)
+    } catch (e) {
+      // address does not ressolve to identity
+    }
+  }
+  const handleClick = () => {
+    showLocalIdentityModal(address)
+      .then(handleResolve)
+      .catch(e => {
+        /* user cancelled modify intent */
+      })
+  }
+  React.useEffect(() => {
+    handleResolve()
+    const subscription = updates$.subscribe(updatedAddress => {
+      if (updatedAddress.toLowerCase() === address.toLowerCase()) {
+        handleResolve()
+      }
+    })
+    return () => subscription.unsubscribe()
+  }, [])
```

Comment:
Same suggestion than for Finance, but I notice now that we should also add `name` as `handleResolve` is using it:

```suggestion
  }, [address, name])
```

Maybe `handleResolve` could take `name` as a parameter, to make it more explicit?

Reply:
Are you sure? I think we just need to pass the address. The name – which in turn is set as the label – is the result of the async side-effect.

Reply:
Oh you’re right, I thought `name` was coming from somewhere else. What could change would be `resolve` though, but we might not want to react to `resolve` changes.